Dockerfile

```diff
@@ -1,46 +1,56 @@
-FROM docker.io/library/rockylinux:9@sha256:d7be1c094cc5845ee815d4632fe377514ee6ebcf8efaed6892889657e5ddaaa6
-LABEL maintainer="Red Hat"
+FROM docker.io/library/rockylinux:9@sha256:d7be1c094cc5845ee815d4632fe377514ee6ebcf8efaed6892889657e5ddaaa6 as rockylinux9
+FROM docker.io/library/golang:1.20.0-bullseye as golang_120
```

Comment:
Why not `golang:1.20.14-bullseye` 1.20.0 is quite old

Reply:
To be fair, 1.21.0 is old too but there's a reason for the X.Y.0th release choices - admittedly though, it only makes sense with 1.21+ where that allows us to fetch literally **any** newer toolchain user may require in their project and cache it as a module properly too (it's a consistent behaviour where a correct toolchain is always downloaded and we don't let Go figure out on its own whether it's bundled toolchain is new enough to process the project) .

The choice for 1.20.0 was mostly for consistency reasons with the 1.21.0 reasoning above and that's pretty much it; it also should be the toolchain used in minimum cases as 1.20 has been EOL'd already and so I plan on ditching it soon in follow-up PRs, so up to you, if you prefer pre-installing a newer 1.20 release (this would happen in a standalone commit though, just like with NodeJS to remain true to the status quo in the multi-stage conversion commit) before ditching it altogether, I can do it.

Reply:
Ok, I see now. Adam also told me, that we need older version of go to dynamically download newer toolchains